utils/stall_free.hh

```diff
@@ -165,7 +165,7 @@ future<> clear_gently(T&) noexcept {
 
 template <HasClearGentlyMethod T>
 future<> clear_gently(T& o) noexcept {
-    return futurize_invoke(std::bind(&T::clear_gently, &o));
+    return futurize_invoke(std::bind(&T::clear_gently, const_cast<std::remove_const_t<T>*>(&o)));
```

Comment:
Is this legal? If it's const, how do we know we're allowed to clear it gently?

Reply:
I guess this is in preparation to destroying it, which is allowed for a const object.

Reply:
> Is this legal? If it's const, how do we know we're allowed to clear it gently?

Right, when the const object is contained in a container, clearing (gently) the container destroys the objects contained in it.

Reply:
I accept it, but I'm uneasy about it. constness is a way to protect an object from mutation. And here we're just ignoring it.

Can we adjust individual templates? For example clear_gently(lw_shared_ptr<const T>) makes sure it is the last owner so it can const_cast the payload.

Are there any dissimilar examples?